teps/0048-task-results-without-results.md

```diff
@@ -283,6 +284,49 @@ spec:
       value: $(tasks.propagate-image-name.results.image)
 ```
 
+#### Users using the Task should feel confident
+
+In the current scenario, there can be a case where a `Task` which claims to produce the result(s) but doesn't produces and the execution of `Task` is successful. This can confuse the user of the `Task`. If the user uses a `Task` then he/she should feel confident that the following `Task` claiming to produce the result is going to produce else it should fail. For example, the following Task fails as the result neither has any default value nor the step produces results.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: add-task
+spec:
+  params:
+    - name: first
+      description: the first operand
+    - name: second
+      description: the second operand
+  results:
+    - name: sum
+      description: the sum of the first and second operand
+  steps:
+    - name: add
+      image: alpine
+      env:
+        - name: OP1
+          value: $(params.first)
+        - name: OP2
+          value: $(params.second)
+      command: ["/bin/sh", "-c"]
+      args:
+        - echo -n $((${OP1}+${OP2}))
+```
+
+whereas doing the following changes in above `Task` will result in success:
+
+```yaml
+...
+spec:
+  ...
+  results:
+    - name: sum
```

Comment:
could we use another example that's not sum, which has a reasonable default value that can't be confused for an actual computed value?

such as repository branch, with default being main